### PR TITLE
[Merged by Bors] - sql: beacons

### PR DESCRIPTION
**Pull request by nkryuchkov**
## Motivation

Part of #2918 

## Changes

- move beacons storage to SQLite
- format SQL code
- extract `*sql.Database` from `mesh`

## Test Plan

UT, ST

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by nkryuchkov** [comment]
bors try

**Comment by dshulyak** [comment]
i meant to extract db from mesh, and reuse it for different modules. instead of creating accessor, could you initialize db in cmd/node instead and pass the pointer to mesh and other modules?

**Comment by nkryuchkov** [comment]
@dshulyak Do you mean passing a `*mesh.DB` to `beacon` and other packages?

**Comment by dshulyak** [comment]
@nkryuchkov no, i was talking about *sql.Database. we can open it in the cmd/node and then pass it to all modules (mesh, beacon, activations, proposals, etc)

File: sql/migrations/0001_initial.sql [hunk 1]
```diff
@@ -1,57 +1,72 @@
...
+CREATE INDEX transaction_by_origin ON transactions (origin, layer);
+CREATE INDEX transaction_by_destination ON transactions (destination, layer);
+
+CREATE TABLE beacons
+(
+    epoch  INT NOT NULL PRIMARY KEY,
+    beacon CHAR(4)
+);
+
+CREATE INDEX beacons_by_epoch ON beacons (epoch);
```
Comment (dshulyak) [inline sql/migrations/0001_initial.sql]: primary key is effectively is an index with unique a constraint, there is no need to add one more

File: sql/migrations/0001_initial.sql [hunk 2]
```diff
@@ -1,57 +1,72 @@
...
-CREATE INDEX transaction_by_origin ON transactions(origin, layer);
-CREATE INDEX transaction_by_destination ON transactions(destination, layer);
+CREATE INDEX transaction_by_origin ON transactions (origin, layer);
+CREATE INDEX transaction_by_destination ON transactions (destination, layer);
+
+CREATE TABLE beacons
+(
+    epoch  INT NOT NULL PRIMARY KEY,
+    beacon CHAR(4)
+);
```
Comment (dshulyak) [inline sql/migrations/0001_initial.sql]: you can add WITHOUT ROWID. in this case it makes sense

File: sql/migrations/0001_initial.sql (hunk 1 above)
Comment (nkryuchkov) [reply sql/migrations/0001_initial.sql]: Indeed, thanks. Removed

File: sql/migrations/0001_initial.sql (hunk 2 above)
Comment (nkryuchkov) [reply sql/migrations/0001_initial.sql]: done

File: mesh/meshdb.go [hunk 3]
```diff
@@ -41,15 +41,8 @@ type DB struct {
...
-	if err := os.MkdirAll(path, os.ModePerm); err != nil {
-		return nil, fmt.Errorf("failed to create %s: %w", path, err)
-	}
-	db, err := sql.Open("file:" + filepath.Join(path, "state.sql"))
-	if err != nil {
-		return nil, fmt.Errorf("open sqlite db %w", err)
-	}
-
+func NewPersistentMeshDB(db *sql.Database, blockCacheSize int, logger log.Log) (*DB, error) {
+	var err error
```
Comment (dshulyak) [inline mesh/meshdb.go]: can we remove it and use err below in if statement? seemed like it hangs in some weird way

File: mesh/meshdb.go [hunk 4]
```diff
@@ -137,6 +130,11 @@ func (m *DB) Transactions() database.Getter {
 	return &txFetcher{mdb: m}
 }
 
+// Beacons exports the beacon database.
+func (m *DB) Beacons() database.Store {
```
Comment (dshulyak) [inline mesh/meshdb.go]: is it a leftover? i think it needs to be removed

File: mesh/meshdb.go [hunk 5]
```diff
@@ -658,3 +656,22 @@ func LayerIDs(db database.Database, namespace string, lid types.LayerID, f func(
 	}
 	return database.ErrNotFound
 }
+
+type beaconDB struct {
```
Comment (dshulyak) [inline mesh/meshdb.go]: and this too

**Review by dshulyak — APPROVED** [review]

File: mesh/meshdb.go (hunk 3 above)
Comment (nkryuchkov) [reply mesh/meshdb.go]: Sure, good idea

File: mesh/meshdb.go (hunk 4 above)
Comment (nkryuchkov) [reply mesh/meshdb.go]: I've already removed it, but after you started reviewing

File: mesh/meshdb.go (hunk 5 above)
Comment (nkryuchkov) [reply mesh/meshdb.go]: already removed, but after I requested your review

**Comment by nkryuchkov** [comment]
bors merge

**Comment by nkryuchkov** [comment]
bors merge

**Comment by nkryuchkov** [comment]
bors merge

**Pull request closed without merging**